libs/hdf-converters/src/xccdf-results-mapper.ts

```diff
@@ -278,142 +299,149 @@ export class XCCDFResultsMapper extends BaseConverter {
+                  _.get(
+                    parseXml(description),
+                    'GroupDescription',
+                    description
+                  ) as string
               },
-              rule_id: {path: 'Rule.id'},
-              check: {path: 'Rule.check'},
-              fix_id: {path: 'Rule.fix.id'},
+              rule_id: {path: 'id'},
+              check: {path: 'check'},
+              fix_id: {path: 'fix.id'},
               fixtext_fixref: {
-                path: 'Rule.fixtext.fixref'
+                path: ['fixtext.fixref.text', 'fixtext.fixref']
               },
-              ident: {path: 'Rule.ident'},
-              reference: {path: 'Rule.reference'},
-              selected: {path: 'Rule.selected'},
-              version: {path: 'Rule.version.text'},
-              weight: {path: 'Rule.weight'},
-              profiles: {
-                path: '$.Benchmark.Profile',
-                transformer: (
-                  profiles: Record<string, unknown>[]
-                ): Record<ProfileKey, unknown>[] => {
-                  const pathsSelect = ['select'];
-                  const paths = {
-                    id: ['id'],
-                    description: ['description.text', 'description'],
-                    title: ['title.text', 'title']
-                  };
-                  return getProfiles(profiles, pathsSelect, paths);
+              ident: {path: 'ident'},
+              reference: {path: 'reference'},
+              selected: {path: 'selected'},
+              version: {path: ['version.text', 'id']},
+              weight: {path: 'weight'},
+              profiles: [
+                {
+                  path: ['profiles'],
+                  id: {path: ['id']},
+                  description: {
+                    path: ['description.text', 'description'],
+                    transformer: (description: string): string =>
+                      _.get(
+                        parseXml(description),
+                        'ProfileDescription',
+                        description
+                      ) as string
+                  },
+                  title: {path: ['title.text', 'title']}
                 }
-              },
+              ],
               rule_result: {
-                path: '$.Benchmark.TestResult',
-                transformer: (testResult: Record<string, unknown>): unknown =>
-                  getRuleResultItem(testResult, RULE_RESULT_PATHS)
+                path: ['ruleResult']
               },
               value: {
-                path: '$.Benchmark.Value',
-                transformer: (values: Record<string, unknown>[]): unknown => {
-                  return _.find(values, (value: Record<string, unknown>) => {
-                    const id = _.get(value, 'id');
-                    return id && id === valueIdTracker;
-                  });
-                }
+                path: ['values']
```

Comment:
this seems to get highly borked at times

for "configure auditd max_log_file_action upon reaching maximum log size"

![image](https://github.com/mitre/heimdall2/assets/2015821/e95eeccf-85d3-4bbe-9231-c2bdbe111017)

![image](https://github.com/mitre/heimdall2/assets/2015821/f152eebe-f916-47d9-9232-83e2794d09de)

Seems like it's extracting out the 'br' stuff from being embedded in the text content and putting it separate whereas the original document is using it as html-encoded formatting :(

Reply:
This problem appears to be caused by shortcoming the XML parser used by this project: https://github.com/NaturalIntelligence/fast-xml-parser/issues/558

I don't know what, if anything, I can do about this one... any suggestions?

Reply:
not considered a blocker - same as the other fast-xml-parser comment